base_business_document_import/models/business_document_import.py

```diff
@@ -506,12 +505,9 @@ def _match_partner_bank(
                 _("IBAN <b>%s</b> is not valid, so it has been ignored.") % iban
             )
             return False
-        company_id = self._context.get("force_company") or self.env.company.id
         bankaccount = rpbo.search(
-            [
-                "|",
-                ("company_id", "=", False),
-                ("company_id", "=", company_id),
+            self._match_company_domain()
+            + [
                 ("sanitized_acc_number", "=", iban),
                 ("partner_id", "=", partner.id),
             ],
```

Comment:
Can't we use `odoo.osv.expression.AND(self._match_company_domain, […])`?

Reply:
If we do that, it should be done uniformly across the file, to stay consistent.
YAGNI.